src/platform/packages/shared/kbn-apm-synthtrace-client/src/lib/utils/generate_id.ts

```diff
@@ -8,14 +8,22 @@
  */
 
 let seq = 0;
-const pid = String(process.pid);
+// reset the sequence every 1_000_000 to avoid overflow
+const MAX_SEQ = 1_000_000;
 
 const LONG_ID_LENGTH = 32;
 const SHORT_ID_LENGTH = 16;
 
 function generateId(length: number = LONG_ID_LENGTH) {
-  const id = String(seq++);
-  const generatedId = pid + id.padStart(length - pid.length, '0');
+  const seqId = String(seq++ % MAX_SEQ);
+  const timestamp = Date.now().toString(16);
+  // additional entropy to avoid collisions
+  const randomPart = Math.floor(Math.random() * Number.MAX_SAFE_INTEGER).toString(16);
```

Comment:
can't use Math.random() here, ids should be consistent (wrt the order that they were created in). Ideally we don't use `process.pid` either but `workerId` instead if it's available, and otherwise we use `0` as a seed

Reply:
also I assume Date.now() will make it too slow

Reply:
> can't use Math.random() here, ids should be consistent (wrt the order that they were created in). Ideally we don't use process.pid either but workerId instead if it's available, and otherwise we use 0 as a seed

we're still using `seqId`.  doesn't that ensure the order they were created in?

Reply:
ah, I see what you mean.

Reply:
`Date.now()` takes around 0.5ms. I think it could be used as one of the variables to create entropy. I guess I just need an extra variable here to ensure uniqueness.

Reply:
Where are you seeing id collision?

Reply:
ok, so maybe we should make it unique in live mode, but not in historical mode?

Reply:
>  I'd like to make sure it is deterministic in all other cases, because I want things to be repeatable as much as possible, so we don't run into issues with snapshots etc.

Makes sense

Reply:
https://github.com/elastic/kibana/pull/214637/commits/f2d4d87dbe1d2e17a97a5674b769d618f87cb0d1